src/main.rs

```diff
@@ -0,0 +1,116 @@
+use std::{
+    env, fs,
+    io::{stderr, Write},
+    path::Path,
+    process,
+};
+
+use getopts::Options;
+use libc::{c_int, getchar, putchar};
+
+fn usage(prog: &str) -> ! {
+    let path = Path::new(prog);
+    let leaf = path
+        .file_name()
+        .map(|x| x.to_str().unwrap_or("ykbf"))
+        .unwrap_or("ykbf");
+    writeln!(&mut stderr(), "Usage: {} [-h] <file.bf>", leaf).ok();
+    process::exit(1)
+}
+
+fn compile(txt: &str) -> (String, Vec<usize>) {
+    // BF comments are anything that's not a command character, so scrub all non-command
+    // characters.
+    let mut out = String::with_capacity(txt.len());
+    for c in txt.chars() {
+        match c {
+            '>' | '<' | '+' | '-' | '.' | ',' | '[' | ']' => {
+                out.push(c);
+            }
+            _ => (),
+        }
+    }
+    out.shrink_to_fit();
+
+    // Pre-calculate the [...] jumps as a map from PC -> PC.
+    let mut map = Vec::with_capacity(out.len());
+    map.resize(out.len(), 0);
+    let mut stack = Vec::new();
+    for (i, c) in out.chars().enumerate() {
+        if c == '[' {
+            stack.push(i);
+        } else if c == ']' {
+            let s = stack.pop().unwrap();
+            map[s] = i;
+            map[i] = s;
+        }
+    }
+
+    (out, map)
+}
+
+fn interp(prg: &[u8], map: Vec<usize>) {
+    let mut pc = 0;
```

Comment:
Can we add comments to these vars.

`pc` is obvious, but I'm not sure what `ptr` or `store` are.

Reply:
I could call them `ptr` and `tape` perhaps? I must admit, I think they're fairly obvious if you know a Turing tape / BF (and if you don't know BF, would comments help?).

Reply:
Let's place this on hold until I've read about bf.

Reply:
I've skimmed the wikipedia page for bf and it uses mixed terminology. I don't feel strongly to be honest, but if you did change your mind, maybe `cells`?

Reply:
(instead of store)

Reply:
OK.